kolena/_experimental/object_detection/evaluator.py

```diff
@@ -0,0 +1,343 @@
+from kolena._experimental.object_detection.utils import threshold_key
+from kolena._experimental.object_detection.workflow import ClassMetricsPerTestCase
+from kolena._experimental.object_detection.workflow import GroundTruth
+from kolena._experimental.object_detection.workflow import Inference
+from kolena._experimental.object_detection.workflow import TestCase
+from kolena._experimental.object_detection.workflow import TestCaseMetrics
+from kolena._experimental.object_detection.workflow import TestSample
+from kolena._experimental.object_detection.workflow import TestSampleMetrics
+from kolena._experimental.object_detection.workflow import TestSuite
+from kolena._experimental.object_detection.workflow import TestSuiteMetrics
+from kolena._experimental.object_detection.workflow import ThresholdConfiguration
+from kolena._experimental.object_detection.workflow import ThresholdStrategy
+from kolena.workflow import Evaluator
+from kolena.workflow import Plot
+from kolena.workflow.metrics import match_inferences_multiclass
+from kolena.workflow.metrics._geometry import InferenceMatches
+from kolena.workflow.metrics._geometry import MulticlassInferenceMatches
+
+
+class ObjectDetectionEvaluator(Evaluator):
+    # Assumes that the first test case retrieved for the test suite contains the complete sample set to be used for
+    # F1-Optimal threshold computation. Subsequent requests for a given threshold strategy (for other test cases) will
+    # hit this cache and use the previously computed population level confidence thresholds.
+    threshold_cache: Dict[str, Dict[str, float]] = {}  # configuration -> label -> threshold
+
+    # Keeps track of test sample locators for each test case (used for total # of image count in aggregated metrics)
+    locators_by_test_case: Dict[str, List[str]] = {}
+
+    def compute_image_metrics(
+        self,
+        ground_truth: GroundTruth,
+        inference: Inference,
+        configuration: ThresholdConfiguration,
+        labels: Set[str],  # the labels being tested in this test case
+    ) -> TestSampleMetrics:
+        assert configuration is not None, "must specify configuration"
+        thresholds = self.get_confidence_thresholds(configuration)
+        bbox_matches: MulticlassInferenceMatches = match_inferences_multiclass(
+            ground_truth.bboxes,
+            [
+                inf
+                for inf in inference.bboxes
+                if inf.label in labels and inf.score >= max(configuration.min_confidence_score, thresholds[inf.label])
+            ],
+            ignored_ground_truths=ground_truth.ignored_bboxes,
+            mode="pascal",
+            iou_threshold=configuration.iou_threshold,
+        )
+        tp = [inf for _, inf in bbox_matches.matched if inf.label in labels]
+        fp = [inf for inf in bbox_matches.unmatched_inf if inf.label in labels]
+        fn = [gt for gt, _ in bbox_matches.unmatched_gt if gt.label in labels]
```

Comment:
Is `gt.label in labels` necessary to check? My expectation is that every box in `gt.bboxes` will have its label in `labels`.